plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -138,39 +140,70 @@ compileLiteral = \case
 -- do different things to the inner expression. This one assumes it's a literal, the other one keeps compiling
 -- through it.
 
-data StringExprContentAs = AsBytes | AsText
+data EncodingAction = UndoUtf8 | PreserveUtf8
```

Comment:
I'd still like a better name than UndoUtf8. The name should help answer the following question: Given a string in the source code, say "Ёжище", _what kind of ByteString does it produce_? If it does truncating (like Haskell's ByteString), call it `Truncate`, if it does something else, say something else. If the name itself is not enough, add Haddock.

Reply:
When I conceived this data type I meant it to answer one question: 
**Should the function undo the UTF8 encoding already applied to the byte string or not?**
From this perspective I could have named it `Utf8PreservationBehavour`.  The idea is that if provides 1 bit of information **currently required** by the `stringLiteralBytes` function (I had a first impulse to use `Bool` but this would have been a [boolean blindness](https://luzkan.github.io/smells/boolean-blindness)) This way I am avoiding the premature generalisation. We can gradually make this data type more general as soon as such need arrives (if ever).

Reply:
I don't know what "undo the UTF8 encoding" means. For example, given `"\12345" :: String`, its UTF-8 encoding is `"\227\128\185"`. What's the output of "undo the UTF8 encoding"?

Reply:
"Undo the UTF8 encoding" is an approximation of the more intricate meaning, which is "Recover a 0 .. 255 **subset of codepoints**  after GHC has UTF8 encoded them"

This function is not defined (hence `Maybe` on the RHS) on the Unicode strings that didn't originally contains every code-point in the range 0-255.

I intent to document this function better. Maybe I should rename it to `fromUtf8EncodedBytes` ?

Reply:
So if I understand correctly, what it does is
- if the string contains only 0-255 codepoints, convert each char to a byte
- fail if it contains anything 256 or above

Then you can call it any of these: `SingleBytes`, `BytesOnly`, `SingleBytesOnly`, `ExtendedAscii`, all of which are better and more informative than `UndoUtf8`.

Also

> This function is not defined (hence Maybe on the RHS) on the Unicode strings that didn't originally contain every code-point in the range 0-255.

This needs to be put into a comment near the definition of this type (or you can add a Note so that you can refer to it in other places - you probably want to refer to it in the function that returns `Maybe`).

Reply:
The best name is the one you don't have to invent :) So I have refactored code not to require that parameter.